pallets/currencies/src/fungibles.rs

```diff
@@ -127,6 +128,46 @@ where
 	}
 }
 
+impl<T: Config + pallet_asset_registry::Config> fungibles::metadata::Inspect<T::AccountId> for FungibleCurrencies<T>
+where
+	T::MultiCurrency: fungibles::Inspect<T::AccountId>,
+	<T::MultiCurrency as fungibles::Inspect<T::AccountId>>::AssetId: From<CurrencyIdOf<T>>,
+	<T::MultiCurrency as fungibles::Inspect<T::AccountId>>::Balance: Into<BalanceOf<T>> + From<BalanceOf<T>>,
+	WithdrawConsequence<BalanceOf<T>>:
+		From<WithdrawConsequence<<T::MultiCurrency as fungibles::Inspect<T::AccountId>>::Balance>>,
+	T::NativeCurrency: fungible::Inspect<T::AccountId>,
+	<T::NativeCurrency as fungible::Inspect<T::AccountId>>::Balance: Into<BalanceOf<T>> + From<BalanceOf<T>>,
+	WithdrawConsequence<BalanceOf<T>>:
+		From<WithdrawConsequence<<T::NativeCurrency as fungible::Inspect<T::AccountId>>::Balance>>,
+	CurrencyIdOf<T>: Into<<T as pallet_asset_registry::Config>::AssetId>,
+{
+	fn name(asset: Self::AssetId) -> Vec<u8> {
+		// Prefer registry for metadata; fall back to sensible defaults
+		if let Some(name) =
+			<pallet_asset_registry::Pallet<T> as hydradx_traits::registry::Inspect>::asset_name(asset.into())
```

Comment:
i dont like that fact that the currencies pallet directly depends on asset registry pallet. 

usually, what we try to avoid is having pallets depeending on each other - unless there is a good reason for it.

Instead, you can just use the hydradx_traits::registry::Inspect trait. That's exactly what is the trait for. You basically asking for something that implements the trait and currencies pallet does not care what that thing is ( now you are asking for specific implemenation - asset registry pallet ).

and you provide the implementation of this trait where it is used.